### PR TITLE
fix: 修复element.get("canvas") 为空时，调用 animate 会报错

### PR DESCRIPTION
**Pull request by Me-Momo**
### 🤔 This is a ...

- [x] Bug fix
- [x] testcase added
- [x] migrate tslint to eslint, and auto lint:fix

### 🔗 Related issue link

related：https://github.com/antvis/G2/issues/3111

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: packages/g-base/src/interfaces.ts [hunk 1]
```diff
@@ -514,14 +514,6 @@ export interface IContainer extends IElement {
...
 }
 
-export interface IGroup extends IContainer {
-  /**
-   * 是否是实体分组，即对应实际的渲染元素
-   * @return {boolean} 是否是实体分组
-   */
-  isEntityGroup(): boolean;
-}
-
```
Comment (hustcc) [inline packages/g-base/src/interfaces.ts:524]: 这个为啥删除？

File: packages/g-canvas/src/util/parse.ts [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
 
-const regexTags = /[MLHVQTCSAZ]([^MLHVQTCSAZ]*)/gi;
-const regexDot = /[^\s\,]+/gi;
-const regexLG = /^l\s*\(\s*([\d.]+)\s*\)\s*(.*)/i;
-const regexRG = /^r\s*\(\s*([\d.]+)\s*,\s*([\d.]+)\s*,\s*([\d.]+)\s*\)\s*(.*)/i;
+const regexLG = /^l\s*\(\s*([\d.]+)\s*)\s*(.*)/i;
+const regexRG = /^r\s*\(\s*([\d.]+)\s*,\s*([\d.]+)\s*,\s*([\d.]+)\s*)\s*(.*)/i;
 const regexPR = /^p\s*\(\s*([axyn])\s*\)\s*(.*)/i;
-const regexColorStop = /[\d.]+:(#[^\s]+|[^\)]+\))/gi;
+const regexColorStop = /[\d.]+:(#[^\s]+|[^)]+\))/gi;
```
Comment (hustcc) [inline packages/g-canvas/src/util/parse.ts:7]: 这个改动是什么？

**Comment by hustcc** [comment]
eslint 升级到最新，我看 ci 还是报错的。

File: packages/g-canvas/src/util/parse.ts (hunk 2 above)
Comment (Me-Momo) [reply packages/g-canvas/src/util/parse.ts:7]: eslint 报错说，不必要的转义 "\"

File: packages/g-base/src/interfaces.ts (hunk 1 above)
Comment (Me-Momo) [reply packages/g-base/src/interfaces.ts:524]: 重复了，定义了多个 IGroup 看下上下文

**Comment by hustcc** [comment]
eslint 升级到最新，我看 ci 还是报错的。

File: .eslintrc [hunk 3]
```diff
@@ -20,64 +25,33 @@
...
-  "parser": "babel-eslint",
   "parserOptions": {
-    "sourceType": "module"
+    "sourceType": "module",
+    "ecmaVersion": 6,
+    "impliedStrict": true
   },
-  "plugins": ["html"],
+  "plugins": ["html", "prettier", "@typescript-eslint", "import"],
   "rules": {
```
Comment (Me-Momo) [inline .eslintrc:34]: rules 基本上跟随 g2plot 除了底部三条 warn 规则

File: packages/g-base/src/abstract/element.ts [hunk 4]
```diff
@@ -453,6 +453,9 @@ abstract class Element extends Base implements IElement {
    *   delay    动画延迟时间
    */
   animate(...args) {
+    if (!this.get('canvas')) {
```
Comment (Me-Momo) [inline packages/g-base/src/abstract/element.ts:456]: 修复问题

File: tsconfig.json [hunk 5]
```diff
@@ -38,7 +38,7 @@
     // "paths": {},                         /* A series of entries which re-map imports to lookup locations relative to the 'baseUrl'. */
     // "rootDirs": [],                      /* List of root folders whose combined content represents the structure of the project at runtime. */
     // "typeRoots": [],                     /* List of folders to include type definitions from. */
-    // "types": [],                         /* Type declaration files to be included in compilation. */
+    "types": ["node"],                         /* Type declaration files to be included in compilation. */
```
Comment (Me-Momo) [inline tsconfig.json:41]: 参考：https://www.labeightyfour.com/2019/05/17/compiler-errors-with-mocha-and-typescript/

**Pull request merged**